src/clarity_vm/tests/contracts.rs

```diff
@@ -324,3 +326,496 @@ fn test_get_block_info_eval_v210() {
         );
     });
 }
+
+fn publish_contract(
+    bc: &mut ClarityBlockConnection,
+    contract_id: &QualifiedContractIdentifier,
+    contract: &str,
+    version: ClarityVersion,
+) -> Result<(), clarity::vm::clarity::Error> {
+    bc.as_transaction(|tx| {
+        let (ast, analysis) =
+            tx.analyze_smart_contract(contract_id, version, contract, ASTRules::PrecheckSize)?;
+        tx.initialize_smart_contract(contract_id, version, &ast, contract, None, |_, _| false)?;
+        tx.save_analysis(contract_id, &analysis)?;
+        Ok(())
+    })
+}
+
+/// Test that you cannot invoke a 2.05 contract
+///  with a trait parameter using a stored principal.
+#[test]
+fn trait_invocation_205_with_stored_principal() {
+    let mut sim = ClarityTestSim::new();
+    sim.epoch_bounds = vec![0, 3, 5];
+
+    // Advance two blocks so we get to Stacks 2.05.
+    sim.execute_next_block(|_env| {});
+    sim.execute_next_block(|_env| {});
+
+    let trait_contract = "(define-trait simple-method ((foo (uint) (response uint uint)) ))";
```

Comment:
To make test more readable, consider extending contract names (same for tests below). 
Ex. "simple-trait" -> "simple-trait-contract" 
"use-simple" -> "use-simple-contract"

Reply:
I'm going to push back on this one. I don't think I see the benefit of this change. I'd prefer not to encourage naming every contract with the `-contract` suffix, since the way it is used should already indicate that it must be a contract.